test/functional/wallet_avoidreuse.py

```diff
@@ -342,14 +342,14 @@ def test_full_destination_group_is_preferred(self):
         txid = self.nodes[1].sendtoaddress(address=ret_addr, amount=0.5)
         inputs = self.nodes[1].getrawtransaction(txid, 1)["vin"]
```

Comment:
Hmm, I added a line to grab the amount and fee for this transaction here, i.e. 
```py
txinfo = self.nodes[1].gettransaction(txid)
print("Paid fee of {} when sending {}".format(txinfo["fee"], txinfo["amount"]))
```
It looks like the fee would be 0.00136966BTC :scream: I thought it might get caught by `maxapsfee` but it went through. Am I missing something else that would catch this case?

I understand this scenario isn't typical, but is this maybe... not the best default option for users? 0.00137 is like $66, and they could be paying this fee on a 0.001BTC transaction?

Reply:
I don't think this is a problem (or at least not worse than any alternative). The high fees are intrinsic in the UTXOs that are already owned by the wallet. The fact that they get realized when making this 0.001BTC payment is no better/worse than if they were realized when making a subsequent payment for 100BTC, or if they were used to make a subsequent payment of 0.00001 BTC.

The only way this could be improved is if our coin selection algorithm favored using relatively few coins when the prevailing fee rate is high, and favored using relatively many coins when the prevailing fee rate is low. I don't believe we have such logic in our wallet.

Reply:
(typed this before github showed johns anwer so there is some duplication here I think) It's not great in theory. But I think we can expect that users know what they are getting into here. The users have reused addresses a lot (willingly or unwillingly) and realized that this is a threat to their privacy. Then they have opted in to avoid reuse which warns users that this can create higher fees via high numbers of inputs. And there isn't even much of a risk here if somebody sets this option just for the fun of it. As long as there is no address reuse in that wallet, avoid reuse also won't kick in.

Because this is a privacy feature that users have opted into it seems the better trade-off to me that someone is surprised by a higher fee rather than being surprised that they could only consolidate 10 outputs in one transaction when they had 30 outputs in that destination. And then they have to further watch out to no compromise themselves when consolidating the remaining 20 outputs. What users are getting aside from privacy protection is a consolidation transaction. While the fees might be high on this first tx it will hopefully also create a change output that is larger and will be much cheaper to spend later on.

`maxapsfee` is only relevant if avoid partial spends is not activated. If avoid partial spends is not set, the wallet will still try to avoid partial spends and prefer to use those coins that avoid partial spends of those that don't if the premium is below the threshold.

Reply:
Ahh thanks for explaining this. I can see how a user turning `-avoidpartialspends` on would be consciously choosing to opt for privacy over lower short-term fees.

> maxapsfee is only relevant if avoid partial spends is not activated. If avoid partial spends is not set, the wallet will still try to avoid partial spends and prefer to use those coins that avoid partial spends of those that don't if the premium is below the threshold.

Ohh my bad, thanks for being patient with me!